packages/abi-coder/src/types/JsonAbiNew.ts

```diff
@@ -0,0 +1,101 @@
+/**
+ * Types for Fuel JSON ABI Format as defined on:
+ * https://github.com/FuelLabs/fuel-specs/blob/master/src/abi/json-abi-format.md
+ */
+export interface JsonAbi {
+  readonly specVersion: string;
+  readonly encodingVersion: string;
+  readonly programType: string;
+  readonly concreteTypes: readonly ConcreteType[];
+  readonly metadataTypes: readonly MetadataType[];
+  readonly functions: readonly AbiFunction[];
+  readonly loggedTypes: readonly LoggedType[];
+  readonly messagesTypes: readonly MessageType[];
+  readonly configurables: readonly Configurable[];
+}
+
+export interface ConcreteType {
+  readonly type: string;
```

Comment:
Is this the human readable name with the full path? 

It would be nice if there were a built in helper function for getting the name without the path (e.g. using a custom type rather than string).

Reply:
Yes, this would be e.g. `struct std:vec::Vec<b256>`. The linked metadata type (via `metadataTypeId`) contains the name of the type: `struct std::vec::Vec`. Why would you want to get the name without the path? What additional abi parsing are you doing that isn't covered by `Interface`?

Reply:
I need the name of the type without the full path and without the generics to display input fields in sway playground, for the UI that lets users call contract functions.

Reply:
I need the name of the type without the full path and without the generics to display input fields in sway playground, for the UI that lets users call contract functions. I can write a helper function that does this for myself, but it seems like a useful this for the SDK to provide.

Reply:
That was changed in:
 - https://github.com/FuelLabs/sway/pull/6254

> Removes the flag `--json-abi-with-callpaths` and the behavior is as if it were true. We removed the flag because it is unsafe to produce JSON ABIs without callpaths, so we shouldn't allow it.

You'll have to do some string parsing to extract only the portion you want.

Reply:
Depending on your use case, this can lead to naming collisions since you can't differentiate types with identical names but different namespaces. I'm taking note of this, and we'll consider adding such parsing utility helpers in the future. For now, the quickest way is to parse it yourself, as you said.